campari/AllASPFuncs.py

```diff
@@ -332,19 +320,22 @@ def construct_psf_background(ra, dec, wcs, x_loc, y_loc, stampsize,
         point *= sed
 
     point = point.withFlux(1, bpass)
-    oversampling_factor = 1
-    convolvedpsf = galsim.Convolve(point, psf)
-    stamp = galsim.Image(stampsize*oversampling_factor,
-                         stampsize*oversampling_factor, wcs=galsim_wcs)
+
+    pointing = util_ref.visit
+    SCA = util_ref.sca
+
+    psf_object = PSF.get_psf_object("ou24PSF", pointing=pointing, sca=SCA, size=stampsize, include_photonOps=False)
+    # See run_one_object documentation to explain this pixel coordinate conversion.
+    x_loc = int(np.floor(x_loc + 0.5))
+    y_loc = int(np.floor(y_loc + 0.5))
+
     # Loop over the grid points, draw a PSF at each one, and append to a list.
-    for a, ij in enumerate(zip(x.flatten(), y.flatten())):
+    for a, ij in enumerate(zip(x_SCA.flatten(), y_SCA.flatten())):
         if a % 50 == 0:
-            Lager.debug(f"Drawing PSF {a} of {np.size(x)}")
+            Lager.debug(f"Drawing PSF {a} of {np.size(x_loc)}")
```

Comment:
Above you were using size of `x_sca`, but here `x_loc`.  Why?  They should of course be the same, but it makes the reader thinks that there's perhaps a reason to use versus the other.  Consider define a new variable.  Something like:
`grid_length = len(x_sca)`  (or `np.size`, as you're using here instead of `len`, either is fine)

Reply:
Or `npoints`, which is the name used in the docstring.